node:events:371
      throw er; // Unhandled 'error' event
      ^

Error: Connection lost: The server closed the connection.
    at Protocol.end (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:112:13)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:94:28)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:526:10)
    at Socket.emit (node:events:406:35)
    at endReadableNT (node:internal/streams/readable:1331:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on Connection instance at:
    at Connection._handleProtocolError (C:\Storage\Node\node_modules\mysql\lib\Connection.js:423:8)
    at Protocol.emit (node:events:394:28)
    at Protocol._delegateError (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:398:10)
    at Protocol.end (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:116:8)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:94:28)
    [... lines matching original stack trace ...]
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  fatal: true,
  code: 'PROTOCOL_CONNECTION_LOST'
}
node:events:371
      throw er; // Unhandled 'error' event
      ^

Error: Connection lost: The server closed the connection.
    at Protocol.end (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:112:13)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:94:28)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:526:10)
    at Socket.emit (node:events:406:35)
    at endReadableNT (node:internal/streams/readable:1331:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on Connection instance at:
    at Connection._handleProtocolError (C:\Storage\Node\node_modules\mysql\lib\Connection.js:423:8)
    at Protocol.emit (node:events:394:28)
    at Protocol._delegateError (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:398:10)
    at Protocol.end (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:116:8)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:94:28)
    [... lines matching original stack trace ...]
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  fatal: true,
  code: 'PROTOCOL_CONNECTION_LOST'
}
C:\Storage\Node\node_modules\mysql\lib\protocol\Parser.js:437
      throw err; // Rethrow non-MySQL errors
      ^

Error: ER_BAD_NULL_ERROR: Column 'name' cannot be null
    at Query.Sequence._packetToError (C:\Storage\Node\node_modules\mysql\lib\protocol\sequences\Sequence.js:47:14)
    at Query.ErrorPacket (C:\Storage\Node\node_modules\mysql\lib\protocol\sequences\Query.js:79:18)
    at Protocol._parsePacket (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:291:23)
    at Parser._parsePacket (C:\Storage\Node\node_modules\mysql\lib\protocol\Parser.js:433:10)
    at Parser.write (C:\Storage\Node\node_modules\mysql\lib\protocol\Parser.js:43:10)
    at Protocol.write (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:38:16)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:88:28)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:526:10)
    at Socket.emit (node:events:394:28)
    at addChunk (node:internal/streams/readable:312:12)
    --------------------
    at Protocol._enqueue (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:144:48)
    at Connection.query (C:\Storage\Node\node_modules\mysql\lib\Connection.js:198:25)
    at C:\Storage\Node\index.js:504:16
    at Layer.handle [as handle_request] (C:\Storage\Node\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\Storage\Node\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (C:\Storage\Node\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (C:\Storage\Node\node_modules\express\lib\router\layer.js:95:5)
    at C:\Storage\Node\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (C:\Storage\Node\node_modules\express\lib\router\index.js:341:12)
    at next (C:\Storage\Node\node_modules\express\lib\router\index.js:275:10) {
  code: 'ER_BAD_NULL_ERROR',
  errno: 1048,
  sqlMessage: "Column 'name' cannot be null",
  sqlState: '23000',
  index: 0,
  sql: "INSERT INTO scrap SET `id` = 0, `name` = NULL, `weight` = 1, `place` = NULL, `time` = '2022-08-09T06:45:44.4895291Z', `rsid` = 0, `did` = 0, `bid` = 0"
}
C:\Storage\Node\node_modules\mysql\lib\protocol\Parser.js:437
      throw err; // Rethrow non-MySQL errors
      ^

Error: ER_BAD_NULL_ERROR: Column 'name' cannot be null
    at Query.Sequence._packetToError (C:\Storage\Node\node_modules\mysql\lib\protocol\sequences\Sequence.js:47:14)
    at Query.ErrorPacket (C:\Storage\Node\node_modules\mysql\lib\protocol\sequences\Query.js:79:18)
    at Protocol._parsePacket (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:291:23)
    at Parser._parsePacket (C:\Storage\Node\node_modules\mysql\lib\protocol\Parser.js:433:10)
    at Parser.write (C:\Storage\Node\node_modules\mysql\lib\protocol\Parser.js:43:10)
    at Protocol.write (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:38:16)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:88:28)
    at Socket.<anonymous> (C:\Storage\Node\node_modules\mysql\lib\Connection.js:526:10)
    at Socket.emit (node:events:394:28)
    at addChunk (node:internal/streams/readable:312:12)
    --------------------
    at Protocol._enqueue (C:\Storage\Node\node_modules\mysql\lib\protocol\Protocol.js:144:48)
    at Connection.query (C:\Storage\Node\node_modules\mysql\lib\Connection.js:198:25)
    at C:\Storage\Node\index.js:504:16
    at Layer.handle [as handle_request] (C:\Storage\Node\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\Storage\Node\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (C:\Storage\Node\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (C:\Storage\Node\node_modules\express\lib\router\layer.js:95:5)
    at C:\Storage\Node\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (C:\Storage\Node\node_modules\express\lib\router\index.js:341:12)
    at next (C:\Storage\Node\node_modules\express\lib\router\index.js:275:10) {
  code: 'ER_BAD_NULL_ERROR',
  errno: 1048,
  sqlMessage: "Column 'name' cannot be null",
  sqlState: '23000',
  index: 0,
  sql: "INSERT INTO scrap SET `id` = 0, `name` = NULL, `weight` = 2, `place` = NULL, `time` = '2022-08-09T06:46:10.0389137Z', `rsid` = 0, `did` = 0, `bid` = 0"
}
